frame/evm/precompile/contracts/kton/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2021 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::{Currency, ExistenceRequirement};
+use sha3::Digest;
+use sp_core::{H160, U256};
+use sp_runtime::traits::UniqueSaturatedInto;
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+use core::str::FromStr;
+use darwinia_evm::{AddressMapping, Runner, Trait};
+use darwinia_evm_primitives::Precompile;
+use ethabi::{Function, Param, ParamType, Token};
+use evm::{Context, ExitError, ExitSucceed};
+use sp_std::borrow::ToOwned;
+
+const DEPOSIT_FUNC: &[u8] = b"deposit(address,uint256)";
+const KTON_PRECOMPILE: &str = "0000000000000000000000000000000000000016";
+/// Kton Precompile Contract
+///
+/// The contract address: 0000000000000000000000000000000000000016
+pub struct Kton<T: Trait> {
+	_maker: PhantomData<T>,
+}
+
+impl<T: Trait> Precompile for Kton<T> {
+	// The execute process of Kton precompile, the input data consists of four parts:
+	// 1. Wkton solidity contract deployed address(20 bytes)
```

Comment:
Could be any contract address?

Reply:
This address refers to the wkton contract address deployed in advance on the dvm. If this address is not correct when executing a call in precompile, it will not succeed.